src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,186 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceTest {
+    private StorageDao storageDao;
+    private RegistrationService registrationService;
+    private User user;
+
+    @BeforeEach
+    public void setUp() {
+        storageDao = new StorageDaoImpl();
+        registrationService = new RegistrationServiceImpl(storageDao);
+        user = new User();
+    }
+
+    @Test
+    void registerNullUser_NotOk() {
+        RegistrationException exception = assertThrows(RegistrationException.class, () ->
+                registrationService.register(null));
+
+        assertEquals("User is null.", exception.getMessage());
+    }
+
+    @Test
+    void registerLoginNull_NotOk() {
+        user.setLogin(null);
+        user.setPassword("validPassword");
+        user.setAge(20);
+
+        RegistrationException exception = assertThrows(RegistrationException.class, () ->
+                registrationService.register(user));
+        assertEquals("Login can't be null.", exception.getMessage());
+    }
+
+    @Test
+    void registerPasswordNull_NotOk() {
+        user.setLogin("validLogin");
+        user.setPassword(null);
+        user.setAge(20);
+
+        RegistrationException exception = assertThrows(RegistrationException.class, () ->
+                registrationService.register(user));
+        assertEquals("Password can't be null.", exception.getMessage());
+    }
+
+    @Test
+    void registerUserAgeNull_NotOk() {
+        user.setLogin("validLog");
+        user.setPassword("validPwd");
+        user.setAge(null);
+
+        RegistrationException exception = assertThrows(RegistrationException.class, () ->
+                registrationService.register(user));
+        assertEquals("Age can't be null.", exception.getMessage());
+    }
+
+    @Test
+    void registerPasswordExists_NotOk() throws RegistrationException {
+        User user = new User();
+        user.setLogin("existingUser");
+        user.setPassword("password123");
+        user.setAge(21);
+        storageDao.add(user);
+
+        User newUser = new User();
+        newUser.setLogin("existingUser");
+        newUser.setPassword("password123");
+        newUser.setAge(21);
+
+        RegistrationException exception = assertThrows(RegistrationException.class, () ->
+                registrationService.register(newUser));
+
+        assertEquals("Login already exists.", exception.getMessage());
```

Comment:
You are using the `register` method to add a user for testing the case of an existing login. As per the checklist, you should use low-level methods like `Storage.people.add()` directly instead of `RegistrationService.register()` to add users for testing.